pkg/metrics/metrics.go

```diff
@@ -31,79 +42,98 @@ type Recorder struct {
 	ReportingPeriod time.Duration
 }
 
+var (
+	Once           sync.Once
+	R              *Recorder
+	ErrRegistering error
+)
+
 // NewRecorder creates a new metrics recorder instance
 // to log the PAC PipelineRun related metrics.
 func NewRecorder() (*Recorder, error) {
-	r := &Recorder{
-		initialized: true,
+	Once.Do(func() {
+		R = &Recorder{
+			initialized: true,
 
-		// Default to 30s intervals.
-		ReportingPeriod: 30 * time.Second,
-	}
+			// Default to 10s intervals.
+			ReportingPeriod: 10 * time.Second,
```

Comment:
how did you choose this number?

Reply:
@chmouel oh, that is mistakenly.

Reply:
Actually I was running some pipelineruns to get metrics which were finishing in 20 seconds and original interval period was 30 seconds so injection code wasn't running for last pipelineruns and metric was having 1 only (it was being updated to 0 after all PR has done) that's why I changed it to 10 seconds, but I don't think that in production there would be any PipelineRun which will ends in 20 seconds.